packages/cyberstorm/src/components/PackageList/PackageList.tsx

```diff
@@ -22,81 +24,133 @@ interface Props {
 
 const PER_PAGE = 20;
 
-/**
- * Fetches packages based on props and shows them as a list.
- *
- * TODO: add loading indicator over the PackageCard section. Doing so
- *       properly might require rethinking the structure of things, i.e.
- *       adding another wrapping component.
- */
-export function PackageList(props: Props) {
+export function PackageList(props: PackageListProps) {
   const { categories, deprecated, listingType, nsfw, searchQuery, section } =
     props;
 
   const [order, setOrder] = useState(PackageOrderOptions.Updated);
   const [page, setPage] = useState(1);
+  const [packageCount, setPackageCount] = useState<number | null>(null);
 
-  const deferredCategories = useDeferredValue(categories);
-  const deferredDeprecated = useDeferredValue(deprecated);
-  const deferredNsfw = useDeferredValue(nsfw);
-  const deferredOrder = useDeferredValue(order);
-  const deferredPage = useDeferredValue(page);
-  const deferredSearchQuery = useDeferredValue(searchQuery);
-  const deferredSection = useDeferredValue(section);
-
-  const deferredIncludedCategories = deferredCategories
+  const includedCategories = categories
     .filter((c) => c.selection === "include")
     .map((c) => c.id);
 
-  const deferredExcludedCategories = deferredCategories
+  const excludedCategories = categories
     .filter((c) => c.selection === "exclude")
     .map((c) => c.id);
 
   useEffect(() => {
+    setPackageCount(null);
     setPage(1);
   }, [categories, deprecated, listingType, nsfw, order, searchQuery, section]);
 
-  const dapper = useDapper();
-  const packages = usePromise(dapper.getPackageListings, [
-    listingType,
-    deferredOrder,
-    deferredPage,
-    deferredSearchQuery,
-    deferredIncludedCategories,
-    deferredExcludedCategories,
-    deferredSection,
-    deferredNsfw,
-    deferredDeprecated,
-  ]);
-
   return (
     <div className={styles.root}>
       <div className={styles.top}>
-        <PackageCount
-          page={page}
-          pageSize={PER_PAGE}
-          searchQuery={searchQuery}
-          totalCount={packages.count}
-        />
+        {packageCount ? (
+          <PackageCount
+            page={page}
+            pageSize={PER_PAGE}
+            searchQuery={searchQuery}
+            totalCount={packageCount}
+          />
+        ) : (
+          <SkeletonBox className={styles.packageCountSkeletonBar} />
+        )}
 
         <PackageOrder order={order} setOrder={setOrder} />
       </div>
 
       <div className={styles.packages}>
-        {packages.results.map((p) => (
-          <PackageCard key={`${p.namespace}-${p.name}`} package={p} />
-        ))}
+        <Suspense
+          fallback={range(1, 13).map((x) => (
```

Comment:
Why 13? Could we maybe use `PER_PAGE` here to avoid page jumping around when user uses the pagination at the bottom? (Obviously it still happens on the last page but we can't know how many items there will be so it can't be helped.)